src/core/Raycaster.js

```diff
@@ -93,9 +93,17 @@ Object.assign( Raycaster.prototype, {
 
 	},
 
-	intersectObject: function ( object, recursive ) {
+	intersectObject: function ( object, recursive, optionalTarget ) {
 
-		var intersects = [];
+		var intersects = optionalTarget || [];
+
+		// Clear target array if specified.
```

Comment:
I think I would prefer if the clearing logic was in the application (ie. aframe) layer instead.